src/plots/cartesian/tick_defaults.js

```diff
@@ -42,7 +42,18 @@ module.exports = function handleTickDefaults(containerIn, containerOut, coerce,
 
             if(!tickFormat && axType !== 'date') {
                 coerce('showexponent', showAttrDflt);
-                coerce('exponentformat');
+
+                var expBase = coerce('exponentbase'),
+                    expFmtDflt = coerce('exponentformat');
+
+                if (expBase !== 2 && expBase !== 10) expFmtDflt = 'power';
+                if (expBase === 2 && expFmtDflt !== 'SI') expFmtDflt = 'power';
+                if (expBase === 'e'){
+                    containerOut.exponentbase = Math.E;
+                    containerOut.type = 'log';
+                }
+
+                containerOut.exponentformat = expFmtDflt;
```

Comment:
:tiger2: let's add a `describe` block for `handleTickDefaults` testing this logic in [`axes_test.js`](https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/axes_test.js#L255).